src/modes/ModeEstimation.jl

```diff
@@ -309,8 +292,7 @@ function optim_function(
     else
         OptimizationFunction(
             l;
-            grad = (G,x,p) -> obj(nothing, G, nothing, x),
-            hess = (H,x,p) -> obj(nothing, nothing, H, x),
```

Comment:
Yes. It does not work - currently we explicitly throw an error if a non-`nothing` Hessian `H` is provided. So it's much more straightforward (and simpler for us) to let Optim etc. handle this case (throw an error, do some fancy AD ...) and just define `f(F, G, x)` instead of `f(F, G, H, x)` (in the Optim API) and only provide `grad` but not `hessian` (in the SciML API).

Reply:
@torfjelde Do you think we should keep it?

Reply:
Ah gotcha, if it doesn't actually currently work then I'm happy to drop it :+1: